src/blockdata/transaction.rs

```diff
@@ -239,6 +250,23 @@ pub enum SubField {
     MysteriousMinerGate(String),
 }
 
+impl Display for SubField {
+    fn fmt(&self, fmt: &mut Formatter<'_>) -> Result<(), FmtError> {
+        match self {
+            SubField::TxPublicKey(PublicKey) => writeln!(fmt, "TxPublicKey: {}", PublicKey),
+            SubField::Nonce(nonce) => {
+                // /let mut cursor = Cursor::new(nonce);
+                let nonce_str = hex_encode(serialize(nonce));
+                writeln!(fmt, "Nonce: {}", nonce_str)
+            }
+            SubField::Padding(padding) => writeln!(fmt, "Padding: {}", padding),
+            SubField::MergeMining(code, hash) => writeln!(fmt, "Merge mining: {}, {}", code, hash),
+            SubField::AdditionalPublickKey(_) => writeln!(fmt, "AdditionalPublickKeys: "),
```

Comment:
This will not print anything after `:`. We should print the `Vec<>` directly

Reply:
Done.